cmd/nerdctl/save.go

```diff
@@ -17,106 +17,124 @@
+                defer f.Close()
+                out = f
+        } else {
+                if isatty.IsTerminal(os.Stdout.Fd()) {
+                        return fmt.Errorf("cowardly refusing to save to a terminal. Use the -o flag or redirect")
+                }
+        }
+        return saveImage(images, out, saveOpts, cmd)
 }
 
 func saveImage(images []string, out io.Writer, saveOpts []archive.ExportOpt, cmd *cobra.Command) error {
-	client, ctx, cancel, err := newClient(cmd)
-	if err != nil {
-		return err
-	}
-	defer cancel()
-
-	allPlatforms, err := cmd.Flags().GetBool("all-platforms")
-	if err != nil {
-		return err
-	}
-	platform, err := cmd.Flags().GetStringSlice("platform")
-	if err != nil {
-		return err
-	}
-	platMC, err := platformutil.NewMatchComparer(allPlatforms, platform)
-	if err != nil {
-		return err
-	}
-
-	saveOpts = append(saveOpts, archive.WithPlatform(platMC))
-
-	imageStore := client.ImageService()
-	for _, img := range images {
-		named, err := referenceutil.ParseAny(img)
-		if err != nil {
-			return err
-		}
-		saveOpts = append(saveOpts, archive.WithImage(imageStore, named.String()))
-	}
-
-	return client.Export(ctx, out, saveOpts...)
+        client, ctx, cancel, err := newClient(cmd)
+        if err != nil {
+                return err
+        }
+        defer cancel()
+
+        allPlatforms, err := cmd.Flags().GetBool("all-platforms")
+        if err != nil {
+                return err
+        }
+        platform, err := cmd.Flags().GetStringSlice("platform")
+        if err != nil {
+                return err
+        }
+        platMC, err := platformutil.NewMatchComparer(allPlatforms, platform)
+        if err != nil {
+                return err
+        }
+
+        saveOpts = append(saveOpts, archive.WithPlatform(platMC))
+
+        imageStore := client.ImageService()
+        var locaImageNames []string
+        walker := &imagewalker.ImageWalker{
+                Client: client,
+                OnFound: func(ctx context.Context, found imagewalker.Found) error {
+                        if found.Image.Name != "" {
+                                locaImageNames = append(locaImageNames, found.Image.Name)
+                        }
+                        return nil
+                },
+        }
+        for _, img := range images {
+                walker.Walk(ctx, img)
+                if len(locaImageNames) == 0 {
+                        named, err := referenceutil.ParseAny(img)
+                        if err != nil {
+                                return err
+                        }
+                        saveOpts = append(saveOpts, archive.WithImage(imageStore, named.String()))
+                } else {
+                        for _, localName := range locaImageNames {
+                                saveOpts = append(saveOpts, archive.WithImage(imageStore, localName))
+                        }
+                }
+        }
+        return client.Export(ctx, out, saveOpts...)
```

Comment:
you need to call `Export` for every matched image string

Reply:
if there are two images have same id, i will export all of them